### PR TITLE
add custom feedback with mandatory input

### LINKED ISSUE #411: I would like to prevent empty description in feedback
**Is your feature request related to a problem? Please describe.**
Right now, a user can provide a feedback with an empty description. The issue created will have a bad output.

**Describe the solution you'd like**
Would be nice to prevent an empty description  by making it mendatory

### PR DESCRIPTION
**Pull request by AlexandreMarkus**
closes #411 

| New feedback output  | Select the feedback type  |  The button is activated once all fields are completed | 
|---|---|---|
| image | image | image |

### CODE REVIEW

File: android/gradle.properties [hunk 1]
```diff
@@ -2,3 +2,7 @@ org.gradle.jvmargs=-Xmx1536M -XX:+UseParallelGC
 android.useAndroidX=true
 android.enableJetifier=true
 extra-gen-snapshot-options=--obfuscate
+org.gradle.daemon=true
+org.gradle.parallel=true
+org.gradle.configureondemand=true
+org.gradle.caching=true
```
Comment (MysticFragilist) [inline android/gradle.properties:8]: those are to improve compilation time for android?

File: lib/ui/widgets/custom_feedback.dart [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    }.toString();
+  }
+
+  Map<String, dynamic> toMap() {
+    return <String, dynamic>{
+      'feedback_type': feedbackType.toString(),
+      'feedback_text': feedbackText,
+    };
+  }
+}
```
Comment (MysticFragilist) [inline lib/ui/widgets/custom_feedback.dart:29]: This could go in the models folder I think 😄

File: lib/ui/widgets/custom_feedback.dart [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+/// What type of feedback the user wants to provide.
+enum FeedbackType {
+  bugReport,
+  featureRequest,
+}
```
Comment (MysticFragilist) [inline lib/ui/widgets/custom_feedback.dart:35]: Idem for this in the constants folder!

**Review by MysticFragilist — CHANGES_REQUESTED** [review]
Really niice implementation! 😄 If i may, for the type of the issue maybe we could use a more user-friendly name in the dropdown instead of having it camelCase. It should also be internationalized. What do you think?

File: android/gradle.properties (hunk 1 above)
Comment (AlexandreMarkus) [reply android/gradle.properties:8]: Yes, exactly

File: lib/ui/widgets/custom_feedback.dart (hunk 2 above)
Comment (AlexandreMarkus) [reply lib/ui/widgets/custom_feedback.dart:29]: Yes totally

**Comment by AlexandreMarkus** [comment]
> Really niice implementation! 😄 If i may, for the type of the issue maybe we could use a more user-friendly name in the dropdown instead of having it camelCase. It should also be internationalized. What do you think?

Yes, good idea! I had some issues with the internationalization. It seems I can't use the AppIntl.of(context), feel free to help ;) . I will try to change this for sure!

File: lib/ui/widgets/custom_feedback.dart (hunk 3 above)
Comment (AlexandreMarkus) [reply lib/ui/widgets/custom_feedback.dart]: added string instead

File: lib/l10n/intl_en.arb [hunk 4]
```diff
@@ -152,6 +152,11 @@
 
   "more_about_applets_title": "About ApplETS",
   "more_report_bug": "Report a bug",
+  "more_report_bug_steps_title": "How does is work?\n\n",
```
Comment (MysticFragilist) [inline lib/l10n/intl_en.arb:155]:
```suggestion
  "more_report_bug_steps_title": "How does it work?\n\n",
```

**Review by MysticFragilist — CHANGES_REQUESTED** [review]
One small typo otherwise would be good to go! Also it seems like there is no test that check the feedbackview_1 golden file. I don't know if it's on purpose, but there is also a displayed flutter red screen error on this golden don't know if a test is missing:
image

**Comment by AlexandreMarkus** [comment]
> One small typo otherwise would be good to go! Also it seems like there is no test that check the feedbackview_1 golden file. I don't know if it's on purpose, but there is also a displayed flutter red screen error on this golden don't know if a test is missing: image

Oups, I forgot to add back the test file, and I corrected the golden ^^

**Review by MysticFragilist — APPROVED** [review]
LGTM! 🚀 Good job separating the feedback from the more_view 😀

**Review by camillebrulotte — APPROVED** [review]
LGTM, Good job, bonne idée ! :D

**Pull request merged**